### PR TITLE
Replaced SASL/PLAIN with OAUTH as recommended authentication mechanism.

### PR DESCRIPTION
**Pull request by AedinC**
Preview available [here](http://file.ams.redhat.com/aecollin/JK-9922/README.html).

### CODE REVIEW

**Comment by k-wall** [comment]
Proposed changes look good to me.

File: docs/kafka/kafka-bin-scripts-kafka/README.adoc [hunk 1]
```diff
@@ -135,21 +135,24 @@ To enable the Kafka scripts to access a Kafka instance, you must configure a con
...
 [source,subs="+quotes"]
 ----
-sasl.mechanism=PLAIN
+sasl.mechanism=OAUTHBEARER
 security.protocol=SASL_SSL
 
-sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
-  username="__<client_id>__" \
-  password="__<client_secret>__" ;
+sasl.oauthbearer.token.endpoint.url=https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token
```
Comment (jbyrne-redhat) [inline docs/kafka/kafka-bin-scripts-kafka/README.adoc:147]: There's an attribute we can use for the token URL. It's `{sso-token-url}`.

**Review by jbyrne-redhat — APPROVED** [review]
Updates look good to me, based on the configuration I see in https://issues.redhat.com/browse/MGDSTRM-9922.

One question: 

Has the updated quick start been tested? We might want to consider a QE verification, or at least a verification from an SME acting in that capacity.

**Comment by AedinC** [comment]
> Updates look good to me, @AedinC, based on the configuration I see in https://issues.redhat.com/browse/MGDSTRM-9922.
> 
> One question:
> 
> Has the updated quick start been tested? We might want to consider a QE verification, or at least a verification from an SME acting in that capacity.

Thanks John. I tested it yesterday and it worked but I can ask QE to test too if you think it's needed.

**Comment by jbyrne-redhat** [comment]
> > Updates look good to me, @AedinC, based on the configuration I see in https://issues.redhat.com/browse/MGDSTRM-9922.
> > One question:
> > Has the updated quick start been tested? We might want to consider a QE verification, or at least a verification from an SME acting in that capacity.
> 
> Thanks John. I tested it yesterday and it worked but I can ask QE to test too if you think it's needed.

Ah, that might be overkill, then. It would take us the rest of the sprint just to have QE test it. If you've tested it, and Keith is happy, then that's probably sufficient!

File: docs/kafka/kafka-bin-scripts-kafka/README.adoc (hunk 1 above)
Comment (AedinC) [reply docs/kafka/kafka-bin-scripts-kafka/README.adoc:147]: Thanks - will change that.

**Pull request merged**